### PR TITLE
[Aptos Framework] [Account] deleted old auth key functions

### PR DESCRIPTION
**Pull request by chloeqjz**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3382)

### CODE REVIEW

**Review by davidiw — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/account.move [hunk 1]
```diff
@@ -183,11 +184,7 @@ module aptos_framework::account {
         *&borrow_global<Account>(addr).authentication_key
     }
 
-    public entry fun rotate_authentication_key(account: &signer, new_auth_key: vector<u8>) acquires Account {
```
Comment (movekevin) [inline aptos-move/framework/aptos-framework/sources/account.move:186]: Why delete this now?

Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/account.move:186]: we had planned to... it is extremely unsafe... we are using signed authorization in move to validate rotation.

**Review by movekevin — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/account.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/account.move:186]: Okay

**Review by davidiw — APPROVED** [review]
muahahah, great

**Review by movekevin — APPROVED** [review]

**Pull request merged**